src/modules/keyboardmanager/ui/EditShortcutsWindow.cpp

```diff
@@ -182,6 +183,9 @@ void createEditShortcutsWindow(HINSTANCE hInst, KeyboardManagerState& keyboardMa
             settingsMessage.Foreground(Windows::UI::Xaml::Media::SolidColorBrush{ Windows::UI::Colors::Red() });
             settingsMessage.Text(winrt::to_hstring("All remappings were not successfully applied."));
         }
+
+        // Save the updated shortcuts remaps to file.
+        std::thread(&KeyboardManagerState::SaveConfigToFile, &keyboardManagerState).detach();
```

Comment:
Since this is on a separate thread, what happens if the user presses Apply, then modifies a mapping and presses Apply again before the save to JSON completed?

Reply:
In that case the second apply operation will fail to save those settings. I will update the logic to use named mutex to handle the json file in that case we will prevent this scenario. We can also update the status of the apply action on the UI in case when save settings operation failed.

Reply:
@arjunbalgovind I have removed the logic to run the saveconfig part to separate thread for now it was making the implementation more complicated. I have tried saving maximum remaps for keys and shortcuts that we can currently have because of non-scrollable feature on UI I haven't noticed any delay on the UI while Apply those changes. We can add this later on if required.